common/org.wso2.developerstudio.eclipse.artifact.security/src/org/wso2/developerstudio/eclipse/security/project/ui/form/SecurityFormPage.java

```diff
@@ -2442,4 +2489,182 @@ public void updateUI(String source) {
 			msg.open();
 		}
 	}
-}
\ No newline at end of file
+
+	/**
+	 * <p>
+	 * Security scenario number should provide for this method.
+	 *
+	 * <p>
+	 * This method opens a shell window which shows the description of mentioned security scenario given in the
+	 * parameter list.
+	 *
+	 * @param scenarioNumber
+	 */
+	public void showSecurityScenarioGraphicalContent(int scenarioNumber) {
+
+		String fileName = SecurityFormConstants.IMAGE_PREFIX + scenarioNumber + SecurityFormConstants.IMAGE_POSTFIX;
+		String relativeFilePath = SecurityFormConstants.RELATIVE_FOLDER_PATH + fileName;
+		String shellTitle = SecurityFormConstants.SHELL_WINDOW_TITLE_PREFIX + scenarioNumber;
+		String securityScenarioTitle = getSecurityScenarioTitle(scenarioNumber);
+		if (!isSecurityScenarioBalloonAlreadyExists(shellTitle)) {
+			Shell shell = new Shell(parentShell);
+			Composite parent = new Composite(shell, SWT.NONE);
+			shell.setBackgroundImage(securityScenarioBalloonBackgroundImage);
+			shell.setLayout(new RowLayout());
+			shell.setText(shellTitle);
+
+			GridLayout gridLayout = new GridLayout();
+			gridLayout.numColumns = NUMBER_OF_COLUMNS_IN_SHELL;
+			parent.setLayout(gridLayout);
+
+			Label scenarioTitle = new Label(parent, SWT.NONE);
+			scenarioTitle.setText(shellTitle + SecurityFormConstants.SECURITY_SCENARIO_TITLE_SEPARATOR
+					+ securityScenarioTitle);
+			GridData textGridData = new GridData();
+			textGridData.horizontalAlignment = SWT.CENTER;
+			scenarioTitle.setLayoutData(textGridData);
+
+			Label labelImage = new Label(parent, SWT.NONE);
+			Image scenarioImage = getScenarioImage(relativeFilePath);
+			labelImage.setImage(scenarioImage);
+			labelImage.setToolTipText(shellTitle);
+			GridData imageGridData = new GridData();
+			imageGridData.horizontalAlignment = SWT.CENTER;
+			labelImage.setLayoutData(imageGridData);
+
+			Rectangle monitorBound = Display.getCurrent().getPrimaryMonitor().getBounds();
+			Rectangle shellBound = shell.getBounds();
+
+			shell.setLocation(getXCoordinate(monitorBound, shellBound), getYCoordinate(monitorBound, shellBound));
+			shell.pack();
+			shell.open();
+		} else {
+			existingShell.open();
+		}
+
+	}
+
+	/**
+	 * This method checks whether there is an already opened shell with title @param securityScenarioTitle and returns
+	 * true if it exists or false if it doesn't exist.
+	 * <p>
+	 * If a shell with same title exists, this method also set existingShell variable to that shell.
+	 * 
+	 * @param securityScenarioTitle
+	 * @return
+	 */
+	private boolean isSecurityScenarioBalloonAlreadyExists(String securityScenarioTitle) {
+		int length = parentShell.getShells().length;
```

Comment:
True. I have replaced this method with a different method and changed the above code by the below PR.
Thanks for the comment.
https://github.com/wso2/developer-studio/pull/92/